app/src/main/java/com/example/designpatternstudy/Iterator/CarIterator.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.designpatternstudy.Iterator
+
+class CarIterator(private val list: MutableList<Car>) : Iterator<Car> {
+
+    private var position: Int = 0
+
+    override fun hasNext(): Boolean {
+        if (position >= list.size || list[position] == null) return false;
+        else return true;
```

Comment:
```suggestion
    override fun hasNext(): Boolean = position < list.size
```
list[position] == null 부분은 의미가 없습니다.
list 구현체에 indices 범위를 벗어난 index에 접근하면 OutOfBoundException이 발생합니다.